baremetal/metal3machine_manager.go

```diff
@@ -533,8 +533,29 @@ func (m *MachineManager) Delete(ctx context.Context) error {
 			host.Spec.NetworkData = nil
 			bmhUpdated = true
 		}
-		if host.Spec.Online {
+
+		//	turn on/off bmh depending on AutomatedCleaningMode and fastTrack values
+
+		//	Automated Cleaning | Fast Track	| BMH
+		//	disabled			false 		turn off
+		//	disabled			true 		turn off
+		//	metadata			false 		turn off
+		//	metadata 			true 		turn on
```

Comment:
Can we have unit tests covering these cases? There are already some tests for the `Delete` function, so you should be able to extend these tests and cover all the behaviors: 

https://github.com/metal3-io/cluster-api-provider-metal3/blob/6ccfd3bf9863794e1b74cf408556be6f8809309a/baremetal/metal3machine_manager_test.go#L1420

Reply:
added